universe/ShipDesign.cpp

```diff
@@ -84,6 +84,52 @@ namespace {
         // have an ID yet) is the same as one that has already been added
         // (which does have an ID)
     }
+
+    constexpr unsigned int CHECKSUM_MODULUS = static_cast<unsigned int>(10'000'000);
+
+    template <class C>
+    void CheckSumCombine(unsigned int& sum, const C& c) {
```

Comment:
effectively implements a custom hash function, which I hope is more portable than boost::hash